agent/consul/server.go

```diff
@@ -889,6 +890,33 @@ func NewServer(config *Config, flat Deps, externalGRPCServer *grpc.Server,
 	// to enable RPC forwarding.
 	s.grpcLeaderForwarder = flat.LeaderForwarder
 
+	// Start watching HCP Link resource. This creates a channel that we can use to
+	// start and stop HCP manager when appropriate. This needs to be created after
+	// the GRPC services are set up in order for the resource service client to
+	// function. This uses the insecure grpc channel so that it doesn't need to
+	// present a valid ACL token.
+	//
+	// If this fails, HCP linking will not work, but to avoid crashing Consul, we log
+	// the error and continue on.
+	hcpLinkWatchCh, err := hcpctl.NewLinkWatch(
+		&lib.StopChannelContext{StopCh: shutdownCh},
+		logger.Named("hcp-link-watcher"),
+		pbresource.NewResourceServiceClient(s.insecureSafeGRPCChan),
+	)
+	if err != nil {
+		s.logger.Error("HCP Link watcher failed to start. HCP Link functionality is disabled", "error", err)
```

Comment:
This might not be feasible and also might be something to do when we add the enable/disable config, but do you think it would be possible to make the link controller aware of this error? Then if a link is created through the API, it would go to an error state rather than successful.

Reply:
hmmm interesting... well one thing is we talked about making the HCP manager on the leader be responsible for marking the Link as `LINKED` or `DISCONNECTED` so if we do make that change, then in the scenario you are talking about, the link would just never get a status. I think that would be better than where we currently are, but I agree with what you're saying — an error state for this scenario would be best. Will think about this!

Reply:
We talked about this on Zoom — not going to address this in this PR, but this [followup ticket](https://hashicorp.atlassian.net/browse/CC-7051), which we are doing this sprint also, will be an improvement.